bin/teardown.js

```diff
@@ -2,33 +2,50 @@
 
 const fs = require("fs");
 const childProcess = require("child_process");
+const minimist = require("minimist");
 
 const retryAsync = require("../src/util/retryAsync");
 const { lambdaRoleName, statesRoleName } = require("../src/config/roleNames");
-const { lambdaPolicyArns, statesPolicyArns } = require("../src/config/policy-arn");
 const { iam, lambda, stepFunctions } = require("../src/aws/services");
 const deleteLambdas = require("../src/aws/deleteLambdas");
 const deleteStateMachine = require("../src/aws/deleteStateMachine");
-const deleteRole = require("../src/aws/deleteRole");
-const detachPolicies = require("../src/aws/detachPolicies");
+const teardownRoleByName = require("../src/aws/teardownRoleByName");
 const getStateMachineArn = require("../src/aws/getStateMachineArn");
 const basename = require("../src/util/basename");
+const promptAsyncYesNoAndExec = require("../src/util/promptAsyncYesNoAndExec");
 
-const stateMachineName = process.argv[2];
-// TODO: Specify Lambdas prepended by a given workflow name to delete
-const lambdaNames = fs.readdirSync("lambdas").map(basename);
+const argv = minimist(process.argv.slice(2), {
+  boolean: ["f", "force"],
+  string: ["roles"],
+  default: {
+    roles: "",
+  },
+});
+const stateMachineName = argv._[0];
+const rolesToDelete = argv.roles.split(",").filter((role) => role.length > 0);
 
 if (!stateMachineName) {
   throw new Error("State machine name needs to be provided");
 }
 
-deleteLambdas(lambdaNames)
-  .catch(console.log)
-  .then(() => detachPolicies(lambdaPolicyArns, lambdaRoleName))
-  .then(() => retryAsync(() => deleteRole(lambdaRoleName), 5, 7000, .6))
+// TODO: Specify Lambdas prepended by a given workflow name to delete
+const lambdaNames = fs.readdirSync("lambdas").map(basename);
 
-getStateMachineArn(stateMachineName)
-  .catch(console.log)
-  .then(deleteStateMachine)
-  .then(() => detachPolicies(statesPolicyArns, statesRoleName))
-  .then(() => retryAsync(() => deleteRole(statesRoleName), 5, 7000, .6))
+const main = async () => {
+  const deleteLambdasPromise = deleteLambdas(lambdaNames).catch(console.log);
+  const deleteStateMachinePromise = getStateMachineArn(stateMachineName)
+    .then(deleteStateMachine)
+    .catch(console.log);
+  await Promise.all([deleteLambdasPromise, deleteStateMachinePromise]);
+
+  rolesToDelete.forEach(teardownRoleByName);
+};
+
+if (argv.force || argv.f) {
```

Comment:
Thid is also command line argument parsing, checking if a `--force` or `-f` flag was provided

Reply:
Good, this is just what we were looking to do.